lib/kdump_utils.pm

```diff
@@ -121,7 +135,18 @@ sub activate_kdump {
     my $memory_total = script_output('kdumptool  calibrate | awk \'/Total:/ {print $2}\'');
     my $memory_kdump = $memory_total >= 2048 ? 1024 : 320;
     my $module_name  = y2_module_consoletest::yast2_console_exec(yast2_module => 'kdump', yast2_opts => '--ncurses');
-    assert_screen([qw(yast2-kdump-disabled yast2-kdump-enabled)], 200);
+    my @initial_tags = qw(yast2-kdump-disabled yast2-kdump-enabled);
+    push(@initial_tags,
+        (is_sle('>=15-sp3')) ? 'yast2-kdump-not-supported' : 'yast2-kdump-cannot-read-mem') if (is_xen_pv);
```

Comment:
Does version check makes some difference at this point? I see check on line 143, which has or logic, so i think you can push both needles to initial_tags without version check.

Reply:
[Not supported pop-up](https://openqa.suse.de/tests/5056963#step/kdump_and_crash/30) has been introduced with newer version of yast2-kdump, currently present in sle15sp3 only. 
The older versions of yast2-kdump allowed the user to enable kdump, therefore yast2 would continue with the initialization process and later on reports [memory read error](https://openqa.suse.de/tests/5259622#step/kdump_and_crash/30) as expected. Thus, we have 2 different flows depending on package version.

Yup, I have chosen the easy way instead of doing version check of the yast2-kdump package, which I admit is not ideal.

@dgdavid Do you plan to backport environment detection and related warnings for older sles as well?

Reply:
Hi there!

@mloviska, I guess you mean to backport https://github.com/yast/yast-kdump/pull/118 and https://github.com/yast/yast-yast2/pull/1120. If so, I'm not sure if it is a good idea. I'll discuss with the team anyway to see if it is possible in a safe way (I mean, without introducing unexpected behaviors).

Reply:
Hey David! 

Yup, exactly. As you mentioned it requires yast2 >= 4.3.45, thus not sure what are the upgrade plans for older release for YaST2. Ok, let's see :)

Thanks for your help!

Reply:
Hi again!

Changes mentioned before ([yast-kdump 4.3.3](https://github.com/yast/yast-kdump/pull/118) and [yast2 4.3.45](https://github.com/yast/yast-yast2/pull/1120)) depends on a bigger one done in the core YaST package few months ago for SP3 too (see [yast2 4.3.6](https://github.com/yast/yast-yast2/pull/1062)) to fix the Xen Host/Guest detection while [adapting storage-ng for not displaying the missing grub partition warning when running in a XEN guest](https://github.com/yast/yast-storage-ng/pull/1102).

For that reason, we think that backporting all of them could be a little bit risky and we wouldn't like to break anything relying in the _previous_ [Yast::Arch XEN detection](https://github.com/yast/yast-yast2/blob/6afd67b306705ffb4c01cff3275d64d2e3a8918e/library/general/src/modules/Arch.rb#L357-L421) in oldest SLE releases.

I'm really sorry.


Have a good day!

Reply:
On the contrary, you have assured us to keep clear separation between the flows. As we won't expect the new warning alert on older release hence older yast2 packages. 

Thanks for clarification, have a nice day as well!